flink-table/flink-sql-gateway/src/test/java/org/apache/flink/table/gateway/service/utils/IgnoreErrorThreadFactory.java

```diff
@@ -0,0 +1,44 @@
+/*
+ *  Licensed to the Apache Software Foundation (ASF) under one
+ *  or more contributor license agreements.  See the NOTICE file
+ *  distributed with this work for additional information
+ *  regarding copyright ownership.  The ASF licenses this file
+ *  to you under the Apache License, Version 2.0 (the
+ *  "License"); you may not use this file except in compliance
+ *  with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.apache.flink.table.gateway.service.utils;
+
+import org.jetbrains.annotations.NotNull;
+
+import java.util.concurrent.ThreadFactory;
+
+/** Factory to create the thread that ignores error message. */
+public class IgnoreErrorThreadFactory implements ThreadFactory {
+
+    public static final IgnoreErrorThreadFactory INSTANCE = new IgnoreErrorThreadFactory();
+
+    private IgnoreErrorThreadFactory() {}
+
+    @Override
+    public Thread newThread(@NotNull Runnable r) {
+        Thread thread = new Thread(r);
```

Comment:
Would be better if can refer to the `ExecutorThreadFactory` add the thread group and thread name for your purpose?

Reply:
Thanks to point out the `ExecutorThreadFactory`. I think it's better we can reuse the code rather than introduce a new class that is almost same. So I remove the `IgnoreErrorThreadFactory` and use the `ExecutorThreadFactory` instead.